subprojects/model-core/src/main/java/org/gradle/internal/reflect/PropertyAccessorType.java

```diff
@@ -115,6 +117,10 @@ public static PropertyAccessorType of(Method method) {
         return null;
     }
 
+    private static boolean isGroovy3() {
+        return GroovySystem.getVersion().startsWith("3."); // FIXME total hack
```

Comment:
Do we want to make this less of a _total_ hack while we're here?

Reply:
Heh, yes.  The real reason for this is that our utility class `o.g.u.i.VersionNumber` is not available in the `model-core` subproject, and this small check doesn't warrant shuffling any classes or dependencies around.  I'll update the comment to reflect this.